src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs

```diff
@@ -12,10 +13,17 @@ public class AzureServiceBusQueueHealthCheck
         : IHealthCheck
     {
         private const string TEST_MESSAGE = "HealthCheckTest";
+        private const string TEST_SESSIONID = "TestSessionId";
         private static readonly ConcurrentDictionary<string, QueueClient> _queueClientConnections = new ConcurrentDictionary<string, QueueClient>();
 
         private readonly string _connectionString;
         private readonly string _queueName;
+
+        /// <summary>
+        /// Null value for this field specifies not being able to connect to queue yet to know session requirement.
+        /// This could happen if queue is not yet created or gets deleted, after health check starts.
+        /// </summary>
+        private bool? _requiresSession;

```

Comment:
private members does not need summary comments

Reply:
who is set _requiresSession? probably you need to add new parameter into AddXX to include this new option?